feature/movie-detail/src/main/java/io/filmtime/feature/movie/detail/MovieDetailViewModel.kt

```diff
@@ -42,8 +43,29 @@ class MovieDetailViewModel @Inject constructor(
       }
 
       is Failure -> {
-        // TODO add in depth error handling
-        _state.value = _state.value.copy(message = result.error.toString(), isLoading = false)
+        when (result.error) {
+          is GeneralError.ApiError -> {
+            _state.value = _state.value.copy(
+              error = result.error,
+              message = (result.error as GeneralError.ApiError).message,
+              isLoading = false,
+            )
+          }
+
+          GeneralError.NetworkError -> {
+            _state.value = _state.value.copy(
+              error = result.error,
+              message = "No internet connection. Please check your network settings.",
```

Comment:
Maybe we have to find a better solution to extract strings and move them in resources. But for now we can live with it